### PR TITLE
feat(cxx_indexer): distinguish direct function calls

### PR DESCRIPTION
**Pull request by zrlk**

### CODE REVIEW

**Comment by zrlk** [comment]
In partial service of #5379 

There are other AST nodes that need supported (like constructor calls).

File: kythe/cxx/indexer/cxx/GraphObserver.h [hunk 1]
```diff
@@ -738,6 +738,14 @@ class GraphObserver {
     Yes
   };
 
+  /// \brief Determines whether a feature is direct or admits indirection.
+  enum class Directness {
+    /// This feature is indirect (e.g., performs dynamic dispatch).
+    Indirect,
```
Comment (shahms) [inline kythe/cxx/indexer/cxx/GraphObserver.h:744]: Thoughts about:
```
enum class CallDispatch {
  kDefault,
  kDirect,
};
```
As the distinction?

**Review by shahms — COMMENTED** [review]
This looks good, just a minor question on naming.

File: kythe/cxx/indexer/cxx/GraphObserver.h (hunk 1 above)
Comment (zrlk) [reply kythe/cxx/indexer/cxx/GraphObserver.h]: SGTM

**Pull request merged**